network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/failure/FailureRecoveryHandler.kt

```diff
@@ -40,37 +44,44 @@ class FailureRecoveryHandler @Inject constructor(
     private val wgTunnelConfig: WgTunnelConfig,
     private val currentTimeProvider: CurrentTimeProvider,
     private val networkProtectionPixels: NetworkProtectionPixels,
+    private val dispatcherProvider: DispatcherProvider,
 ) : WireguardHandshakeMonitor.Listener {
 
-    private var recoveryCompleted = false
-    private var recoveryInProgress = false
-    override suspend fun onTunnelFailure(lastHandshakeEpocSeconds: Long) {
+    private var failureRecoveryInProgress = false
```

Comment:
This should be an atomic boolean as we're checking the variable in potentially different threads